README.md

```diff
@@ -494,6 +494,45 @@ One major difference FiloDB has from the Prometheus data model is that FiloDB su
 
 FiloDB offers an improved accuracy `histogram_max_quantile` function designed to work with a max column from the source.  If clients are able to send the max value captured during a window, then we can report more accurate upper quantiles (ie 99%, 99.9%, etc.) that do not suffer from clipping.
 
+### Step & Lookback PromQL Improvements
+
+The value provided within square brackets in the Range Vector Selector expression of the PromQL expression
+is commonly known as "range" or "lookback". It indicates how far behind each instant
+the query engine will lookback to normalize the value for any given instant.
+
+**Problem 1:** PromQL has an important gotcha on range value. Hard coded lookback
+can yield to queries with unintended effects. Lets consider the expression `rate(foo[5m])`. The result
+of this query would have completely different meanings with different step values. Since step parameter
+typically is not controlled by user (as in Grafana), it can have silent negative effects. For example, if
+lookback becomes smaller than step because longer time range was chosen, then query silently drops samples.
```

Comment:
Samples are only dropped if the Lookback is below the publish interval, it has nothing to do with step.  Please correct....  or clarify what you mean here.

Reply:
samples are indeed dropped if lookback is less than publish interval. I can explain offline. This has already been documented elsewhere. It is a long explanation, and needs to have a separate context and effort. We need to document how the query engine works in regards to lookback and step.

Reply:
updated docs